pkg/kocache/kocache.go

```diff
@@ -102,18 +151,18 @@ func (c *koCache) GetOrAddEntry(path string) *entry {
 	}
 
 	if e == nil {
-		if c.entries == nil {
-			return nil // was shutdown
+		if !c.centralReady.Load() {
+			return nil, errCentralUnreachable
+		} else if c.entries == nil {
+			return nil, errKoCacheShuttingDown
+		} else {
+			e = newEntry()
+			c.entries[path] = e
+			go e.Populate(c.onlineCtx, c.upstreamClient, fmt.Sprintf("%s/%s", c.upstreamBaseURL, path), c.opts)
```

Comment:
Curious that no tests accused a data race here. Accessing `c.onlineCtx` should be synchronized wiht `ctxMutex` since you could swap the underlying pointer when `GoOnline()` is called.

Reply:
I actually had the lock here and then I removed it to check if the race tests are passing and they did :) 
Or maybe this is something that still can happen and the race tests are not catching?
Is context copied by value when calling `Populate`? I think, it is not...

Reply:
It still passes `go test -race -count 50 -p 4 -v ./pkg/kocache/... ` :)